src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -430,149 +453,125 @@ private LoadDocumentEntity LoadDocument(PostedDataEntity postedData, bool loadAl
                 }
                 loadDocumentEntity.SetPages(pageData);
             }
+
             loadDocumentEntity.SetGuid(documentGuid);
             loadDocumentEntity.SetShowGridLines(globalConfiguration.Viewer.GetShowGridLines());
+            
             // return document description
             return loadDocumentEntity;
         }
 
-
-        private dynamic GetHandler()
-        {
-            if (viewerHtmlHandler != null)
-            {
-                return viewerHtmlHandler;
-            }
-            else
-            {
-                return viewerImageHandler;
-            }
-        }
-
-        private PageDescriptionEntity GetPageDescriptionEntities(PageData page)
+        private PageDescriptionEntity GetPageDescriptionEntities(Page page)
         {
             PageDescriptionEntity pageDescriptionEntity = new PageDescriptionEntity();
             pageDescriptionEntity.number = page.Number;
-            pageDescriptionEntity.angle = page.Angle;
+            // TODO: because page.Angle does not exist
+            pageDescriptionEntity.angle = 0;
             pageDescriptionEntity.height = page.Height;
             pageDescriptionEntity.width = page.Width;
             return pageDescriptionEntity;
         }
 
-        private string GetPageContent(PageData page, string password, string documentGuid)
+        private string RenderPageToString(int pageNumberToRender, string documentGuid, string password)
         {
-            if (globalConfiguration.Viewer.GetIsHtmlMode())
-            {
-                HtmlOptions htmlOptions = new HtmlOptions();
-                SetOptions(htmlOptions, password, page.Number);
-                // get page HTML              
-                return this.GetHandler().GetPages(documentGuid, htmlOptions)[0].HtmlContent;
+            byte[] bytes;
 
-            }
-            else
+            // TODO: consider adding usage of the RedisCache
+            using (MemoryStream pageStream = RenderPageToMemoryStream(pageNumberToRender, password, documentGuid))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameters to 'RenderPageToMemoryStream' have the same names but not the same order as the method arguments.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5242614)